primefaces-showcase/src/main/webapp/ui/data/datatable/columns.xhtml

```diff
@@ -33,7 +33,7 @@
 
                     <p:columns value="#{dtColumnsView.columns}" var="column" headerText="#{column.header}" field="#{column.property}">
                         <f:facet name="filter">
-                            <p:datePicker selectionMode="range" onchange="PF('customersTable').filter()" rendered="#{column.type eq 'date'}"/>
+                            <p:datePicker selectionMode="range" converter="javax.faces.DateTime" onchange="PF('customersTable').filter()" rendered="#{column.type eq 'date'}"/>
```

Comment:
You didn't have to change any types in the Java?

Reply:
The AJava is using `javax.time.LocalDate`  I have a feeling its when you use `java.util.Date` this happens like we do but it doesn't hurt giving it the converter "hint".